src/Rules/PhpDoc/RequireExtendsCheck.php

```diff
@@ -41,39 +43,48 @@ public function checkExtendsTags(Node $node, array $extendsTags): array
 
 		foreach ($extendsTags as $extendsTag) {
 			$type = $extendsTag->getType();
-			if (!$type instanceof ObjectType) {
+			if (count($type->getObjectClassNames()) === 0) {
 				$errors[] = RuleErrorBuilder::message(sprintf('PHPDoc tag @phpstan-require-extends contains non-object type %s.', $type->describe(VerbosityLevel::typeOnly())))
 					->identifier('requireExtends.nonObject')
 					->build();
 				continue;
 			}
 
-			$class = $type->getClassName();
-			$referencedClassReflection = $type->getClassReflection();
+			$classNames = $type->getObjectClassNames();
```

Comment:
Introduce this var earlier so you don't need to call `getObjectClassNames` twice (methods called on Type can get expensive)